File: db/supa.py
from dotenv import load_dotenv
import psycopg2
import random
import os
from supabase import create_client, Client

def unique_id():
    """
    Generate a unique ID.
    """
    seed = random.getrandbits(32)
    while True:
        yield seed
        seed += 1

def main():

    try:
        load_dotenv()

        USER = os.getenv("user")
        PASSWORD = input("ENTER THE PASSWORD NOW: ")
        HOST = os.getenv("host")
        PORT = os.getenv("port")
        DBNAME = os.getenv("dbname")

        conn = psycopg2.connect(
            user=USER,
            password=PASSWORD,
            host=HOST,
            port=PORT,
            dbname=DBNAME
        )
        print("Connection successful!")
        
        cursor = conn.cursor()
        users_user_id = next(unique_id())
        users_name = input("Enter your name: ")
        users_email = input("Enter your email: ")
        users_username = input("Enter your username: ")
        users_password = input("Enter your password: ")
        users_notifications = input("Do you want notifications? (yes/no): ")
        users_notifications = True if users_notifications.lower() == 'yes' else False
        users_notifications_interval = input("Enter notification interval (in minutes): ")
        sample_data = [
            (users_user_id, users_name, users_email, users_password, users_username, users_notifications, users_notifications_interval),
        ]

        insert_query = """
        INSERT INTO users (userid, name, email, password, username, notificationtoggle, notificationinterval)
        VALUES (%s, %s, %s, %s, %s, %s, %s)
        """

        cursor.executemany(insert_query, sample_data)
        conn.commit()

        print("Sample data inserted!")

        # Cleanup
        cursor.close()
        conn.close()

        read_sample_table()

    except Exception as e:
        print(f"Issue: {e}")

def read_sample_table():
    load_dotenv()

    url: str = os.getenv("SUPABASE_URL")
    key: str = os.getenv("SUPABASE_KEY")

    supabase: Client = create_client(url, key)
    print("READING FROM SAMPLE DATA TABLE:")
    response = supabase.table("users").select("*").execute()

    print(response)
    
if __name__ == "__main__":
    print("Setup and Demo of Environment:\n")
    main()

File: db/supa.js
// index.js
import 'dotenv/config';
import { Client as PGClient } from 'pg';
import readline from 'readline';
import { createClient } from '@supabase/supabase-js';



// simple password prompt (note: input is visible; if you want masking, use a library)
async function promptPassword(promptText = 'ENTER THE PASSWORD NOW: ') {
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
  const question = (q) => new Promise((resolve) => rl.question(q, resolve));
  const answer = await question(promptText);
  rl.close();
  return answer;
}

async function main() {
  try {
    const USER = process.env.user;   // matches your Python env variable names
    const HOST = process.env.host;
    const PORT = process.env.port;
    const DBNAME = process.env.dbname;
    const PASSWORD = await promptPassword('ENTER THE PASSWORD NOW: ');

    if (!USER || !HOST || !PORT || !DBNAME) {
      throw new Error('Missing required Postgres env vars: user, host, port, dbname');
    }

    const pg = new PGClient({
      user: USER,
      host: HOST,
      port: Number(PORT),
      database: DBNAME,
      password: PASSWORD,
    });

    await pg.connect();
    console.log('Connection successful!');

    // Close the PG connection (mirrors your Python)
    await pg.end();

    // Then read via Supabase SDK like your read_sample_table()
    await readSampleTable();

  } catch (e) {
    console.error(`Issue: ${e.message || e}`);
  }
}

async function readSampleTable() {
  const url = process.env.SUPABASE_URL;
  const key = process.env.SUPABASE_KEY;

  if (!url || !key) {
    throw new Error('SUPABASE_URL and SUPABASE_KEY are required in .env');
  }

  const supabase = createClient(url, key);
  console.log('READING FROM SAMPLE DATA TABLE:');

  const { data, error } = await supabase.from('users').select('*');
  if (error) {
    console.error('Supabase read error:', error.message);
  } else {
    console.log(data);
  }
}

console.log('Setup and Demo of Environment:\n');
main();
